activesupport/lib/active_support/local_store.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module LocalStore # :nodoc:
```

Comment:
Maybe I'm too scarred from Active Support development but I keep thinking this is a new `Cache` store. I wonder if we should just call this `State`, `IsolationAwareState` or `IsolatedState`?

Reply:
`LocalState`?